tests/rule_based_profiler/config/test_base.py

```diff
@@ -279,3 +283,65 @@ def test_rule_based_profiler_from_commented_map():
     commented_map = CommentedMap(data)
     config = RuleBasedProfilerConfig.from_commented_map(commented_map)
     assert all(hasattr(config, k) for k in data)
+
+
+def test_resolve_config_using_acceptable_arguments(
+    profiler_with_placeholder_args: RuleBasedProfiler,
+) -> None:
+    config: RuleBasedProfilerConfig = (
+        RuleBasedProfilerConfig.resolve_config_using_acceptable_arguments(
+            profiler=profiler_with_placeholder_args
+        )
+    )
+
+    # The resolved config is equivalent to the old one but is a new instance
+    old_config: RuleBasedProfilerConfig = profiler_with_placeholder_args.config
+    assert id(old_config) != id(config)
+    assert old_config.__dict__ == config.__dict__
```

Comment:
This extends that idea of immutability - the config is equivalent in value but it is an entirely different instance. This allows us to maintain the idealized version of a config (a snapshot of the arguments used to instantiate the business logic object).

Reply:
@cdkini Despite my previous example with the website/database example, this approach is totally fine (and can be better from code maintenance perspective, which is the main thrust of functional programming style).  Thanks. /cc @donaldheppner